ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Simple.hs

```diff
@@ -19,6 +19,14 @@ data TriggerHardFork =
     -- | For testing only, trigger the transition at a specific hard-coded
     -- epoch, irrespective of the ledger state.
   | TriggerHardForkAtEpoch !EpochNo
-    -- | Never trigger a hard fork
-  | TriggerHardForkNever
+    -- | Ledger states in this era cannot determine when the hard fork
+    -- transition will happen.
+    --
+    -- It's crucial to note this era could eventually end via a hard fork
+    -- transition. But that could only happen once it is no longer configured
+    -- to use this setting (eg in a subsequent software release). Therefore,
+    -- this option does /not/ imply that "the era will never end", merely that
+    -- "the era cannot end within this node process before it restarts with
+    -- different software and/or configuration for this era".
+  | TriggerHardForkCannotBeKnown
```

Comment:
Naming bikeshidding: My first thought on seeing the new name was "why is it not `TriggerHardForkUnknown`?". Is there some connotation of `Unknown` you want to avoid that `CannotBeKnown` does not have? I could see differences if this were a "dynamic" value (part of some mutable state), where `CannotBeKnown` feels like it should be a terminal state, but one might have an `Unknown -> Known` state transition. But `TriggerHardFork` is static configuration data.

In your PR description, you write

> See the single commit's message.

Right now, the commit description is empty, maybe that would be a place to clarify?

Reply:
Doh! Thanks for pointing out I forgot the commit message. Fixed.

Reply:
And I changed the name to `TriggerHardForkNotDuringThisExecution`. Verbose, but very concrete.